components/director/internal/domain/runtime/resolver.go

```diff
@@ -51,11 +62,23 @@ func (r *Resolver) Runtimes(ctx context.Context, filter []*graphql.LabelFilter,
 		cursor = string(*after)
 	}
 
+	tx, err := r.db.Begin()
+	if err != nil {
+		return nil, err
+	}
+
+	ctx = r.ctxvs.WithValue(ctx, appdb.DatabaseCtxKey, tx)
```

Comment:
probably we should check if context already has a tx?

Reply:
Why? There will be no error if we want to overwrite value for a context key. And we rather don't want to pass shared context for multiple resolvers in single HTTP request, right?